### PR TITLE
(TWILL-119) Make YarnAppClient supports HA RM

### PR DESCRIPTION
**Pull request by chtyim**
- HA RM is added since Hadoop 2.3.

### CODE REVIEW

File: twill-yarn/src/main/hadoop20/org/apache/twill/internal/yarn/Hadoop20YarnAppClient.java [hunk 1]
```diff
@@ -211,7 +218,6 @@ public YarnApplicationReport getReport() {
       try {
         return new Hadoop20YarnApplicationReport(yarnClient.getApplicationReport(appId));
       } catch (YarnRemoteException e) {
-        LOG.error("Failed to get application report {}", appId, e);
```
Comment (anew) [inline twill-yarn/src/main/hadoop20/org/apache/twill/internal/yarn/Hadoop20YarnAppClient.java]: removing the logging here is good. However, the messages could add useful context (here: the appId). If you wrap it into a RuntimeException, why not add the message?

File: twill-yarn/src/main/hadoop23/org/apache/twill/internal/yarn/Hadoop23YarnAppClient.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <p>
+ * The service implementation of {@link YarnAppClient} for Apache Hadoop 2.1 and beyond.
```
Comment (anew) [inline twill-yarn/src/main/hadoop23/org/apache/twill/internal/yarn/Hadoop23YarnAppClient.java]: 2.3

File: twill-yarn/src/main/hadoop23/org/apache/twill/internal/yarn/Hadoop23YarnAppClient.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ *
+ * The {@link VersionDetectYarnAppClientFactory} class will decide to return instance of this class for
+ * Apache Hadoop 2.1 and beyond.
```
Comment (anew) [inline twill-yarn/src/main/hadoop23/org/apache/twill/internal/yarn/Hadoop23YarnAppClient.java]: ditto

**Comment by chtyim** [comment]
@anew Thanks for the review. Addressed comments, please have another look.

**Pull request closed without merging**